muttdown/main.py

```diff
@@ -27,24 +29,28 @@ def convert_one(part, config):
         text = part.get_payload(decode=True)
         if not isinstance(text, six.text_type):
             # no, I don't know why decode=True sometimes fails to decode.
+            # it is because its decoding from different formats. first base64 then unicode ---guygma
             text = text.decode('utf-8')
         if not text.startswith('!m'):
             return None
         text = re.sub('\s*!m\s*', '', text, re.M)
+        f = tempfile.NamedTemporaryFile(suffix='_panmail.css')
+        if config.css: f.write(config.css)
         if '\n-- \n' in text:
             pre_signature, signature = text.split('\n-- \n')
-            md = markdown.markdown(pre_signature, output_format="html5")
-            md += '\n<div class="signature" style="font-size: small"><p>-- <br />'
-            md += '<br />'.join(signature.split('\n'))
-            md += '</p></div>'
+            message  = pypandoc.convert_text(pre_signature, 'html5', format='md', \
+                    extra_args=["--css="+f.name, "--self-contained", "--metadata=pagetitle:'email'"])
+            message += '\n<div class="signature" style="font-size: small"><p>-- \n<br />'
+            message += '<br />'.join(signature.split('\n'))
+            message += '</p></div>'
         else:
-            md = markdown.markdown(text)
-        if config.css:
-            md = '<style>' + config.css + '</style>' + md
-            md = pynliner.fromString(md)
-        message = MIMEText(md, 'html', _charset="UTF-8")
+            message  = pypandoc.convert_text(text, 'html5', format='md', \
+                    extra_args=["--css="+f.name, "--self-contained", "--metadata=pagetitle:'email'"])
```

Comment:
see above comment regarding correct indentation for multiline function call arguments